### PR TITLE
Added support for CMEK in alloydb cluster and automated backup

### PR DESCRIPTION
**Pull request by ravisiddhu**
Added support for CMEK in alloydb cluster and automated backup.

CMEK is immutable for cluster but it is updatable for automated backup.

This is one of the 2 PR is to fix https://github.com/hashicorp/terraform-provider-google/issues/13250

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

```release-note:enhancement
alloydb: added `encryption_config` and `encryption_info` fields in `google_alloydb_cluster`, to allow CMEK encryption of the cluster's data.
```

```release-note:enhancement
alloydb: added the `encryption_config` field inside the `automated_backup_policy` block in`google_alloydb_cluster`, to allow CMEK encryption of automated backups.
```

### CODE REVIEW

**Review by SarahFrench — COMMENTED** [review]
Hi @ravisiddhu, thanks for your contribution. I've just been taking a look and it seems that the changes you made to automatedBackupPolicy in [this commit](https://github.com/GoogleCloudPlatform/magic-modules/commit/be83a1ff706a051ab7d368ac0b976f2854f7c317) have been lost. Could you re-add that change to your PR please?

**Comment by ravisiddhu** [comment]
> Hi @ravisiddhu, thanks for your contribution. I've just been taking a look and it seems that the changes you made to automatedBackupPolicy in [this commit](https://github.com/GoogleCloudPlatform/magic-modules/commit/be83a1ff706a051ab7d368ac0b976f2854f7c317) have been lost. Could you re-add that change to your PR please?

I have re-added that change.

File: mmv1/products/alloydb/Cluster.yaml [hunk 1]
```diff
@@ -89,6 +89,36 @@ properties:
...
+            - :TYPE_UNSPECIFIED
+            - :GOOGLE_DEFAULT_ENCRYPTION
+            - :CUSTOMER_MANAGED_ENCRYPTION
+          output: true
+        - !ruby/object:Api::Type::Array
+          name: kmsKeyVersions
+          item_type: Api::Type::String
+          description: |
+            Output only. Cloud KMS key versions that are being used to protect the database or the backup.
+          output: true
```
Comment (SarahFrench) [inline mmv1/products/alloydb/Cluster.yaml:121]: Currently the build is failing for this PR, and the solution is to remove one level of indentation in this section of yaml.   Both the `encryptionInfo` and `encryptionConfig` sections should start at the same level of indentation.

Could you please fix the indentation, and then I can run the automated tests on this PR? Thanks!

```suggestion
  - !ruby/object:Api::Type::NestedObject
    name: "encryptionInfo"
    description: |
      EncryptionInfo describes the encryption information of a cluster or a backup.
    output: true
    properties:
      - !ruby/object:Api::Type::Enum
        name: 'encryptionType'
        description: "Output only. Type of encryption."              
        values:
          - :TYPE_UNSPECIFIED
          - :GOOGLE_DEFAULT_ENCRYPTION
          - :CUSTOMER_MANAGED_ENCRYPTION
        output: true
      - !ruby/object:Api::Type::Array
        name: kmsKeyVersions
        item_type: Api::Type::String
        description: |
          Output only. Cloud KMS key versions that are being used to protect the database or the backup.
        output: true
```

Comment (ravisiddhu) [reply mmv1/products/alloydb/Cluster.yaml]: Done

**Review by SarahFrench — APPROVED** [review]

**Pull request merged**